### ISSUE TITLE
Outimage sr.py

### ISSUE THREAD

**Issue by raialvaro** [remicres/sr4rs#18]
Hello again Remi!

I have re-launched a training with this image LR
![image](https://user-images.githubusercontent.com/71493478/112597031-e359d880-8e0c-11eb-9893-03ccaa6d4339.png)

and this image HR
![image](https://user-images.githubusercontent.com/71493478/112597095-f9679900-8e0c-11eb-8a31-3436fc543d91.png)

There are about 1,443 patches.

This is the command for the training and Tensorboar graphics

**python train.py --lr_patches lr_patches.tif --hr_patches hr_patches.tif --vggfile /SR/sr4rs/code/vgg19.npy --save_ckpt /SR/sr4rs/code/ckpts/ --logdir /SR/sr4rs/code/results/logs/ --vggweight 0.00003 --l1weight 200 --l2weight 0.0 --depth 64 --vggfeatures 1234 --batchsize 4 --adam_lr 0.0002 --lr_scale 0.0001 --hr_scale 0.0001 --savedmodel /SR/sr4rs/code/savedmodel --epochs 670**

![image](https://user-images.githubusercontent.com/71493478/112597453-6e3ad300-8e0d-11eb-941b-47d8968fed9a.png)

The training time has lasted 3 days.

Well when I go to predict with Mr.py with the same training image, I don't understand why it shows a bad result like this  for a specific area
![image](https://user-images.githubusercontent.com/71493478/112597779-e1444980-8e0d-11eb-95f4-163cde5b9f9a.png)

Can you tell me what I am doing wrong, or what parameters are important to show a better result?

Thank so much again! I'm sorry insistence

**Comment by remicres** [comment]
Hi @raialvaro ,
It is not that bad :wink: 

1. Check the losses values

Can you tick the box "_ignore outliers in chart scaling_" in Tensorboard? Also, can you add the plots for the L2 loss and the VGG loss?
All the thing is to find some "good" losses weights, so that the losses values are in the same order of magnitude (say after a few epochs).

2. Try with a higher learning rate

When I look the images, I am under the impression that the training has not been completed.
Maybe you can try to increase a bit the learning rate: give a try with 0.02

3. Double check your data (However I feel that this is Ok in your case)

Can you check that your patches covers the same area? I mean, before the patches were sampled, have you carefully re-sampled the images so that the LR image pixel spacing is **exactly 4 times** the HR image spacing? If you open the lr_patches and hr_patches in QGis (or any raster viewer), the patches must cover the exact same area. In a general way, a quick look to your patches enables you to identify some issues, sometimes.

**Comment by raialvaro** [comment]
Hi remi!

1. Check the losses values

I show graphics

![image](https://user-images.githubusercontent.com/71493478/112623715-fa102780-8e2c-11eb-952d-d048f6e04318.png)

![image](https://user-images.githubusercontent.com/71493478/112623415-9d147180-8e2c-11eb-8f16-193b037887b2.png)

I don't understand why L2 has that graph, some parameter training error?

2. Try with a higher learning rate

Thank you! I will try with your advice learning rate 0.002. So, You recommend changing the depth from 64 to 32 with this patches?

3. Double check your data (However I feel that this is Ok in your case)

I already checked that the LR and HR images match perfectly. Specifically, the images are geometrically co-registered so that the pixels match

![image](https://user-images.githubusercontent.com/71493478/112624943-8f5feb80-8e2e-11eb-9207-7db5b7796245.png)

**Comment by remicres** [comment]
Ok, your losses looks good.
Maybe later you could try to decrease a bit L1 and VGG losses weights (because the GAN loss is quite small in your case).

Do not worry about the L2 loss: since it is unused for training, the scalar summary is 0 (even though that is not true!), just ignore it.

I bet it will be Ok with a higher learning rate.

**Comment by raialvaro** [comment]
**python train.py --lr_patches lr_patches.tif --hr_patches hr_patches.tif --vggfile /SR/sr4rs/code/vgg19.npy --save_ckpt /SR/sr4rs/code/ckpts/ --logdir /SR/sr4rs/code/results/logs/ --vggweight 0.00003 --l1weight 200 --l2weight 0.0 --depth 64 --vggfeatures 1234 --batchsize 4 --adam_lr 0.0002 --lr_scale 0.0001 --hr_scale 0.0001 --savedmodel /SR/sr4rs/code/savedmodel**

When you refer to raising the learning rate to 0.002 you mean the parameter hr_scale and lr_scale at the same time, or which one exactly ???

Thanks!!

**Comment by remicres** [comment]
Just the learning rate (`--adam_lr`).

```
python train.py \
--lr_patches lr_patches.tif \
--lr_scale 0.0001 \
--hr_patches hr_patches.tif \
--hr_scale 0.0001 \
--vggfile /SR/sr4rs/code/vgg19.npy \
--save_ckpt /SR/sr4rs/code/ckpts/ \
--logdir /SR/sr4rs/code/results/logs/ \
--vggweight 0.00003 \
--l1weight 200 \
--l2weight 0.0 \
--depth 64 \
--vggfeatures 1234 \
--batchsize 4 \
--adam_lr 0.002 \
--savedmodel /SR/sr4rs/code/savedmodel_new
```

**Comment by raialvaro** [comment]
Great Remi!!

You will have news about the progress!!!

Thank so much!
